### ISSUE TITLE
Readiness probe configured to fire before liveness

### ISSUE THREAD

**Issue by pweil-** [fbladilo/miq-on-openshift#33]
cc @mfojtik

[In the template the readiness probe is configured with an initial delay of 180 seconds and the liveness probe is configured with an initial delay of 300 seconds](https://github.com/fbladilo/miq-on-openshift/blob/master/templates/miq-template.yaml#L92-L103).  This looks backwards to me.  Liveness indicates that the container is running, readiness means the container is ready to serve traffic and should be included in any service that selects the pod. [Description of probes](http://kubernetes.io/docs/user-guide/pod-states/#container-probes)

Was there a particular reason for these settings?

FWIW, when I was testing this locally my readiness probe never succeeded but I haven't done any debugging as to why.

**Comment by fbladilo** [comment]
@pweil- Initially they were both set at 300 seconds but the service (httpd) was ready much earlier than the 300 seconds so I lowered readiness to 180 seconds. What we really need I believe is a different method of liveness probe instead of http get on 80, maybe checking a local file would be enough with a shorter initialdelayseconds. Is a file existance check standard practice for liveness probes?

If the deployment failed check volume permissions, it is usually where I find issues since the miq-app POD runs privileged.

**Comment by pweil-** [comment]
If the container is ready it should also be live.  Failure of the liveness probe would indicate that Kube needs to restart the container...it is dead.  So maybe you don't need the liveness probe at all or, as you say, maybe a different liveness probe would be better. 

I'm not sure about checking the existence of the file unless that file indicates that the container is running but not ready to receive traffic.

**Comment by fbladilo** [comment]
@pweil- @bazulay I think I can have a basic file/dir existance check on vmdb (application root) for liveness, that at least tells me there is a container there and seems alive, then just let the readiness probe be the real test as you mentioned.

**Comment by pweil-** [comment]
I did a little more looking and noticed some of our db templates actually include timing like this where the liveness probe has a delay longer than the readiness probe.  In those cases, liveness checks an open port and readiness is checking that a query can be performed so it is ready to receive traffic.  So I think what you're doing is consistent with that.  

Since liveness is considered SUCCESS before the probe actual fails I think this is fine - the pod becomes ready before liveness is confirmed and liveness can then trigger restarts later.  Closing the issue.

Reference: https://github.com/openshift/origin/blob/e0c8f93be1bf173f5932796c4d3cbd96310b4de7/examples/db-templates/mongodb-ephemeral-template.json#L94-L107
